src/runtime_src/core/common/xclbin_parser.cpp

```diff
@@ -229,6 +229,20 @@ kernel_max_ctx(const ip_data& ip)
   return ctxid;
 }
 
+//Get the cu functional from kernel xml entry
+static uint32_t
+get_functional(const pt::ptree& xml_kernel, const std::string& element)
+{
+  for (auto& elem : xml_kernel) {
+    if (elem.first != element)
+      continue;
+
+    return convert(elem.second.get<std::string>("<xmlattr>.functional"));
```

Comment:
The convert function returns size_t, maybe `static_cast<uint32_t>(convert(...))` just to be explicit.
Without cast may fail on some compiler?

Reply:
Yes, compiler does complain about the data type conversion may lose data.

I just changed it from uint32_t to size_t